vector/src/main/java/im/vector/app/features/home/room/list/home/release/ReleaseNotesPreferencesStore.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.list.home.release
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import org.matrix.android.sdk.api.extensions.orFalse
+import javax.inject.Inject
+
+private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "release_notes")
+
+class ReleaseNotesPreferencesStore @Inject constructor(
+        private val context: Context
+) {
+
+    private val isAppLayoutOnboardingShown = booleanPreferencesKey("SETTINGS_APP_LAYOUT_ONBOARDING_SHOWN")
+
+    val appLayoutOnboardingShown: Flow<Boolean> = context.dataStore.data
+            .map { preferences -> preferences[isAppLayoutOnboardingShown].orFalse() }
+            .distinctUntilChanged()
+
+    suspend fun setAppLayoutOnboardingShown(isShown: Boolean) {
+        context.dataStore.edit { settings ->
+            settings[isAppLayoutOnboardingShown] = isShown
+        }
+    }
```

Comment:
Are you sure the Activity will not be show for user launching Element for the first time? This is a requirement of the issue. I do not see the logic here.

Reply:
You can (maybe) test the nullability of `HomeActivityViewState.authenticationDescription` https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/home/HomeActivityViewState.kt#L25 to have the info.

Reply:
fixed - https://github.com/vector-im/element-android/pull/7003/files#diff-7153f5b6cf4c510293d53c73474e854a3a84e6003c88341472c0808377bf3462R121

also fixed problem when this value was null after login - https://github.com/vector-im/element-android/pull/7003/files#diff-31d1cb8d12c34d852a52405908bc8f1a427a1f08485a662527a6ff5c11e7053eR630

Reply:
> also fixed problem when this value was null after login - https://github.com/vector-im/element-android/pull/7003/files#diff-31d1cb8d12c34d852a52405908bc8f1a427a1f08485a662527a6ff5c11e7053eR630

I think this fixes the issue, but I am wondering if this is the best place to set the `state.selectedAuthenticationState`. Not blocking this PR, but maybe @ouchadam can have a quick look later on this change.

Reply:
for context, the reason why this was `null` previously is because the value was only being used to track _signup_ events in the [HomeActivityViewModel](https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/home/HomeActivityViewModel.kt#L156) 

for the change, I think it's fine, we could eventually remove the `selectedAuthenticationState` parameter and instead rely on updating the state before calling `onSessionCreated`